app_dart/lib/src/service/branch_service.dart

```diff
@@ -63,6 +75,44 @@ class BranchService {
     }
   }
 
+  /// Creates a flutter/recipes branch that aligns to a flutter/flutter branch.
+  ///
+  /// Take the example repo history:
+  ///   flutter/flutter: A -> B -> C -> D -> E
+  ///   flutter/recipes: V -> W -> X -> Y -> Z
+  ///
+  /// If flutter/flutter branches at C, this finds the flutter/recipes commit that should be used for C.
+  /// The best guess for a flutter/recipes commit that aligns with C is whatever was the most recently committed
+  /// before C was committed.
+  ///
+  /// Once the flutter/recipes commit is found, it is branched to match flutter/flutter.
+  ///
+  /// Generally, this should work. However, some edge cases may require CPs. Such as when commits land in a
+  /// short timespan, and require the release manager to CP onto the recipes branch (in the case of reverts).
```

Comment:
Is a manual branch needed for such case?

Reply:
No, but manual CPs will be needed to be applied. Ideally, we need a way to know what commits in flutter/recipes are green, and it might be something we revisit later if other tools need this information as well.